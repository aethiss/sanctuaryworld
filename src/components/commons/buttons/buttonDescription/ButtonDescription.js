import React from 'react'
import PropTypes from 'prop-types'

// Next
import Link from 'next/link'

// Material-ui && Styles
import { makeStyles } from '@material-ui/core/styles'
import { OldFenris } from '../../commonStyles/fontFamily'

const useStyles = makeStyles(() => ({
  button: {
    background: 'url("/images/buttons/button-interactive.png") no-repeat',
    backgroundPosition: '0 0',
    height: '133px',
    width: '280px',
    border: 'none',
    outline: 'none',
    position: 'relative',
    cursor: 'pointer',
    '&:hover': {
      backgroundPosition: '0 -132px',
    }
  },
  buttonText: {
    ...OldFenris,
    color: '#d0c3c3',
    fontSize: '16px',
    position: 'absolute',
    width: '100px',
    marginLeft: '-31px',
    marginTop: '-10px',
  }
}))

const ButtonDescription = ({ text, anchorLink, action }) => {
  const classes = useStyles()

  const renderButton = () => (
    <button
      className={classes.button}
      onClick={action}
    >
      <span className={classes.buttonText}>{text}</span>
    </button>
  )

  return (
    <>
      { anchorLink ? (
        <Link href={anchorLink}>
          <a>
            {renderButton()}
          </a>
        </Link>
      ) : renderButton() }
    </>
  )
}

ButtonDescription.propTypes = {
  text: PropTypes.string,
  anchorLink: PropTypes.string,
  action: PropTypes.func,
}

export default ButtonDescription
