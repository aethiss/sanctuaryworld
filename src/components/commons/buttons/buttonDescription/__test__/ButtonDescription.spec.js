import React from 'react'
import TestRenderer from 'react-test-renderer'
import ButtonDescription from '../ButtonDescription'

describe('ButtonDescription', () => {
  describe('Render', () => {
    it('[ButtonDescription] : Render default', () => {
      const testRenderer = TestRenderer.create(<ButtonDescription />)
      expect(testRenderer).toMatchSnapshot()
    })
    it('[ButtonDescription] : Render anchor', () => {
      const customProps = {
        anchorLink: '/',
        text: 'test',
      }
      const testRenderer = TestRenderer.create(<ButtonDescription {...customProps} />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
