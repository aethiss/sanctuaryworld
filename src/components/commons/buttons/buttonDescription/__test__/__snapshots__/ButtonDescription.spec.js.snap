// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDescription Render [ButtonDescription] : Render anchor 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <button
    className="makeStyles-button-1"
  >
    <span
      className="makeStyles-buttonText-2"
    >
      test
    </span>
  </button>
</a>
`;

exports[`ButtonDescription Render [ButtonDescription] : Render default 1`] = `
<button
  className="makeStyles-button-1"
>
  <span
    className="makeStyles-buttonText-2"
  />
</button>
`;
